code/datums/star_economy.dm

```diff
@@ -0,0 +1,326 @@
+/datum/star_resource
+	var/name = "fire the coders"
+	var/cname = "resource"
+
+	var/scale_price = 100000
+	var/scale_weight = 500
+
+
+/datum/star_resource/iron
+	name = "iron ore"
+	cname = "iron"
+
+	scale_price = 40000
+	scale_weight = 2000
+
+/datum/star_resource/hyperlite
+	name = "hyperlite crystals"
+	cname = "hyper"
+
+	scale_price = 200000
+	scale_weight = 150
+
+/datum/star_resource/silicon
+	name = "silicate ore"
+	cname = "silicon"
+
+	scale_price = 60000
+	scale_weight = 1000
+
+/datum/subsystem/starmap/proc/cname2resource(var/cname)
+	for(var/datum/star_resource/resource in star_resources)
+		if(resource.cname == cname)
+			return resource
+
+/datum/subsystem/starmap/proc/generate_galactic_prices()
+	var/list/galactic_supply = list()
+	var/list/galactic_producers = list()
+	for(var/datum/star_system/system in star_systems)
+		if(!system.primary_station)
+			continue
+		for(var/resource in system.primary_station.resources)
+			if(galactic_supply.Find(resource))
+				galactic_supply[resource] += system.primary_station.resources[resource]
+				galactic_producers[resource] += 1
+			else
+				galactic_supply[resource] = system.primary_station.resources[resource]
+				galactic_producers[resource] = 1
+
+	for(var/datum/star_resource/R in star_resources)
+		var/supply = galactic_supply[R.cname]
+		var/producers = galactic_producers[R.cname]
+
+		if(!supply || !producers)
+			galactic_prices[R.cname] = "ERROR"
+			continue
+
+		galactic_prices[R.cname] = round(R.scale_price / (supply / producers),0.01) //cents only yo
+
+
+/datum/subsystem/starmap/proc/generate_system_prices(var/datum/star_system/system)
+	if(!system.primary_station)
+		return
+	for(var/resource in system.primary_station.resources)
+		var/datum/star_resource/R = cname2resource(resource)
+		if(!R)
+			continue
+		if(resource == system.primary_station.primary_resource)
+			system.primary_station.prices[resource] = round(min(PRICE_CAP,(R.scale_price / system.primary_station.resources[resource])),0.01)
+		else
+			system.primary_station.prices[resource] = round(min(PRICE_CAP,((R.scale_price / system.primary_station.resources[resource] + galactic_prices[resource]) / 2)),0.01)
+
+/datum/subsystem/starmap/proc/generate_faction_prices(var/datum/star_faction/faction)
+	for(var/resource in faction.resources)
+		var/datum/star_resource/R = cname2resource(resource)
+		if(!R)
+			continue
+
+		faction.prices[resource] = round(min(PRICE_CAP,(R.scale_price / faction.resources[resource])),0.01)
+
+
+
+
+/datum/subsystem/starmap/proc/process_economy()
+	if(world.time % 18000 == 1) //every 30 minutes
```

Comment:
*screams*
*SCREAMS*
*AHHH!!*
**THATS NOT HOW YOU...**

You need to use a separate subsystem if you want to do this.